File: backend/src/main/java/com/backend/BenchMarks/model/Comparacao.java
package com.backend.BenchMarks.model;

import java.time.LocalDate;

import com.backend.BenchMarks.model.dto.ComparacaoDTO;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "dados_comparacao")
public class Comparacao {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private LocalDate data;
    
    // Dados da primeira localidade
    private String identificadorLocalidade1;
    private Integer confirmados1;
    private Integer mortes1;
    private Double taxaLetalidade1;
    private Long populacaoEstimada1;
    private Double casosPor100kHab1;
    
    // Dados da segunda localidade
    private String identificadorLocalidade2;
    private Integer confirmados2;
    private Integer mortes2;
    private Double taxaLetalidade2;
    private Long populacaoEstimada2;
    private Double casosPor100kHab2;
    
    // Dados comparativos calculados
    private Integer diferencaConfirmados;
    private Integer diferencaMortes;
    private Double diferencaTaxaLetalidade;
    private Double diferencaCasosPor100kHab;
    
    @ManyToOne
    @JoinColumn(name = "comparacao_dto_id", nullable = false)
    private ComparacaoDTO comparacao;
}

File: frontend/src/app/modules/Comparacao.ts
import { ID } from "@datorama/akita";
import { ComparacaoDTO } from "./dto/ComparacaoDTO";

export class Comparacao{
  id: ID;
  data: Date;
  identificadorLocalidade1: string;
  confirmados1: number;
  mortes1: number;
  taxaLetalidade1: number;
  populacaoEstimada1: number;
  casosPor100kHab1: number;
  identificadorLocalidade2: number;
  confirmados2: number;
  mortes2: number;
  taxaLetalidade2: number;
  populacaoEstimada2: number;
  casosPor100kHab2: number;
  diferencaConfirmados: number;
  diferencaMortes: number;
  diferencaTaxaLetalidade: number;
  diferencaCasosPor100kHab: number;

  comparacao: ComparacaoDTO[];

  constructor(
    id: ID,
    data: Date,
    identificadorLocalidade1: string,
    confirmados1: number,
    mortes1: number,
    taxaLetalidade1: number,
    populacaoEstimada1: number,
    casosPor100kHab1: number,
    identificadorLocalidade2: number,
    confirmados2: number,
    mortes2: number,
    taxaLetalidade2: number,
    populacaoEstimada2: number,
    casosPor100kHab2: number,
    diferencaConfirmados: number,
    diferencaMortes: number,
    diferencaTaxaLetalidade: number,
    diferencaCasosPor100kHab: number,
    comparacao: ComparacaoDTO[]= []
  ) {
    this.id = id;
    this.data = data;
    this.identificadorLocalidade1 = identificadorLocalidade1;
    this.confirmados1 = confirmados1;
    this.mortes1 = mortes1;
    this.taxaLetalidade1 = taxaLetalidade1;
    this.populacaoEstimada1 = populacaoEstimada1;
    this.casosPor100kHab1 = casosPor100kHab1;
    this.identificadorLocalidade2 = identificadorLocalidade2;
    this.confirmados2 = confirmados2;
    this.mortes2 = mortes2;
    this.taxaLetalidade2 = taxaLetalidade2;
    this.populacaoEstimada2 = populacaoEstimada2;
    this.casosPor100kHab2 = casosPor100kHab2;
    this.diferencaConfirmados = diferencaConfirmados;
    this.diferencaMortes = diferencaMortes;
    this.diferencaTaxaLetalidade = diferencaTaxaLetalidade;
    this.diferencaCasosPor100kHab = diferencaCasosPor100kHab;
    this.comparacao = comparacao;

  }

}
